jerry-core/jcontext/jcontext.h

```diff
@@ -116,9 +116,9 @@ typedef struct
   jmem_cpointer_t debugger_byte_code_free_tail; /**< tail of byte code free linked list */
   uint32_t debugger_flags; /**< debugger flags */
   uint16_t debugger_receive_buffer_offset; /**< receive buffer offset */
-  uint16_t debugger_port; /**< debugger socket communication port */
   uint8_t debugger_message_delay; /**< call receive message when reaches zero */
-  int debugger_connection; /**< holds the file descriptor of the socket communication */
+  jerry_debugger_transport_t *debugger_transport_p; /**< holds the pointer to the debugger transport */
+  jerry_debugger_conn_t *debugger_connection_p; /**< holds the pointer to the debugger connection */
```

Comment:
Do we need to separate these two?

Reply:
we can probably keep just the transport obj, and have the connection be part of the transport, @martijnthe ?

Reply:
Yeah, I think so.

I don't see a use case for being able to distinguish the transport and the connection.

I guess in a multi-Jerry-context one could have multiple connections, sharing the same transport implementation. But one could work around that by just creating separate `jerry_debugger_transport_t`s instead I guess.